package/MDAnalysis/core/selection.py

```diff
@@ -1381,3 +1447,46 @@ def _parse_subexp(self):
 
 # The module level instance
 Parser = SelectionParser()
+
+
+def gen_selection_class(singular, attrname, dtype, per_object):
+    """Class factory for arbitrary TopologyAttrs
+
+    Parameters
+    ----------
+    singular: str
+        singular name of TopologyAttr
+    attrname: str
+        attribute name of TopologyAttr
+    dtype:
+        type of TopologyAttr
+    per_object:
+        level of TopologyAttr
+
+    Returns
+    -------
+    Selection: subclass of Selection
+    """
+    basedct = {"token": singular, "field": attrname}
+    name = f"{attrname}Selection"
+    if issubclass(dtype, bool):
+        basecls = BoolSelection
+
+    elif issubclass(dtype, int):
+        basecls = RangeSelection
+
+    elif issubclass(dtype, float):
+        basecls = FloatRangeSelection
+
+    elif issubclass(dtype, str) or dtype == object:
+        basecls = _ProtoStringSelection
+        if per_object == "segment":
+            basedct["level"] = "segindices"
+        elif per_object == "residue":
+            basedct["level"] = "resindices"
+        else:
+            basedct["level"] = "ix"
+    else:
+        raise ValueError(f"No base class defined for dtype {dtype}")
```

Comment:
Who will see this error? If it can occur when a user adds their own TopologyAttrs (something we are mildly encouraging I believe) then there needs to be more guidance on what to do.

Reply:
Updated to warn on TopologyAttr creation (if dtype is not one of the obvious ones) and expanded error message